gadgets/trace_exec/program.bpf.c

```diff
@@ -172,16 +161,17 @@ static __always_inline bool upper_layer()
 		return false;
 	}
 
-	struct _ovl_inode *oi =
-		container_of(inode, struct _ovl_inode, vfs_inode);
-	if (!oi) {
-		return false;
-	}
-
 	struct dentry *upperdentry;
 
+	// struct ovl_inode defined in fs/overlayfs/ovl_entry.h
+	// Unfortunately, not exported to vmlinux.h
+	// and not available in /sys/kernel/btf/vmlinux
+	// See https://github.com/cilium/ebpf/pull/1300
```

Comment:
That would be good. But even if the cilium/ebpf PR is merged, we still have work to do to make use of it in Inspektor Gadget:

* How an image-based gadget can specify which kernel module to load, so that the `run` command uses the correct `KernelTypes`:
```
ebpf.CollectionOptions{
    MapReplacements: mapReplacements,
    Programs: ebpf.ProgramOptions{
        KernelTypes: /* BTF from the kernel and the modules */,
    },
}
```

* How does it interact with BTFGen? #2045